internal/mode/static/telemetry/collector.go

```diff
@@ -147,3 +158,40 @@ func collectGraphResourceCount(
 
 	return ngfResourceCounts, nil
 }
+
+func collectNGFReplicaCount(ctx context.Context, k8sClient client.Reader, podNSName types.NamespacedName) (int, error) {
+	var pod v1.Pod
+	if err := k8sClient.Get(ctx,
+		types.NamespacedName{Namespace: podNSName.Namespace, Name: podNSName.Name},
+		&pod,
+	); err != nil {
+		return 0, err
+	}
+
+	podOwnerRefs := pod.GetOwnerReferences()
+	if podOwnerRefs == nil {
+		return 0, errors.New("could not get owner reference of NGF Pod")
+	}
+	if len(podOwnerRefs) != 1 {
+		return 0, errors.New("multiple owner references of NGF Pod")
+	}
+
+	switch kind := podOwnerRefs[0].Kind; kind {
+	case "ReplicaSet":
```

Comment:
I chatted with @kate-osborn and we had a couple of concerns which were tough to find an answer to, 

1. Would we be in a position where the NGF Pod has multiple OwnerReferneces
2. Would we be in a position where the NGF Pod's OwnerRefernece is a `Deployment` instead of a `ReplicaSet`

If anyone else has any thoughts on these concerns it would be great to hear them or on how we handled them (erroring in those cases)

Reply:
1. I don't know the answer to that. It's a list, which makes it seem like it could be. But I don't think I've seen it before.
2. A Deployment creates a Replicaset, so I don't think we can hit this case.

Reply:
I wonder what other way we could get the number of pods though. Through checking matching labels? I don't know if that's reliable.

Reply:
Yeah, I think this is the best way to get the replicas, I was just pointing out the edge case. But, I think we have it covered with the error checks around the length of the owner refs and the owner kind.